packages/cletus/src/operations/dba.tsx

```diff
@@ -604,6 +604,164 @@ export const data_index = operationOf<
+        const content = await fs.readFile(filePath, 'utf-8');
+        
+        // Process file content in chunks with overlap
+        const chunks = chunkTextWithOverlap(content, chunkSize, overlap);
+        ctx.chatStatus(`  Extracting from ${chunks.length} chunks...`);
+        
+        // Process chunks in parallel batches
+        const batchSize = 5;
+        for (let i = 0; i < chunks.length; i += batchSize) {
+          const batch = chunks.slice(i, Math.min(i + batchSize, chunks.length));
+          const batchResults = await Promise.all(
+            batch.map(async (chunk, batchIdx) => {
+              const chunkIndex = i + batchIdx;
+              try {
+                const extracted = await extractDataFromChunk(ctx, type, typeSchema, chunk);
+                return extracted;
+              } catch (error) {
+                ctx.log(`Warning: Failed to extract from chunk ${chunkIndex + 1}: ${(error as Error).message}`);
+                return [];
+              }
+            })
+          );
+          
+          // Flatten and add to all records
+          allExtractedRecords.push(...batchResults.flat());
+          ctx.chatStatus(`  Processed ${Math.min(i + batchSize, chunks.length)}/${chunks.length} chunks`);
+        }
+      } catch (error) {
+        ctx.log(`Warning: Failed to read file ${filePath}: ${(error as Error).message}`);
+      }
+    }
+    
+    ctx.chatStatus(`Extracted ${allExtractedRecords.length} potential records`);
+    
+    // Get sample records for uniqueness determination
+    const sampleSize = Math.min(10, allExtractedRecords.length);
+    const sampleRecords = allExtractedRecords.slice(0, sampleSize);
+    
+    // Ask AI to determine unique fields
+    let uniqueFields: string[] = [];
+    if (allExtractedRecords.length > 0) {
+      ctx.chatStatus('Determining unique fields...');
+      uniqueFields = await determineUniqueFields(ctx, type, typeSchema, sampleRecords);
+    }
+    
+    ctx.chatStatus(`Using unique fields: ${uniqueFields.length > 0 ? uniqueFields.join(', ') : 'none (all records will be imported)'}`);
+    
+    // Merge data based on unique fields
+    const existingRecords = dataManager.getAll();
+    let imported = 0;
+    let updated = 0;
+    let skipped = 0;
+    const updatedIds: string[] = [];
+    
+    for (const record of allExtractedRecords) {
+      // Find existing record by unique fields
+      let existingRecord = null;
+      if (uniqueFields.length > 0) {
+        existingRecord = existingRecords.find(existing => 
+          uniqueFields.every(field => existing.fields[field] === record[field])
+        );
+      }
+      
+      if (existingRecord) {
+        // Check if there are actual changes
+        const hasChanges = Object.keys(record).some(
+          field => JSON.stringify(existingRecord!.fields[field]) !== JSON.stringify(record[field])
+        );
+        
+        if (hasChanges) {
+          await dataManager.update(existingRecord.id, record);
```

Comment:
Let's turn all of these into a single dataManager.save function call instead of all of these which can be slow reloading the file Everytime.